### ISSUE TITLE
Feature request - File name as note title

### ISSUE THREAD

**Issue by juranta** [pbek/QOwnNotes#157]
With OS X and iOS devices, I keep my notes in a plain text (Markdown) format and manage them with applications such as nvALT, Sublime Text and several others. These programs are quite similar to QOwnNotes. I'd like to use QOwnNotes without sync with Windows for a similar purpose, since I tried it and noticed it's a quite good and robust choice. 

There's one thing that prevented me from using it with my collection of notes. It automatically inserts the top header lines when creating a new note, and it doesn't search from file names. My notes use file name as a "header" of a kind, which is also the main text I use when searching for my notes. Is it reasonable to add an option to QOwnNotes to support this kind of use?

**Comment by pbek** [comment]
I'm not sure I understand what you are seeking. Currently the file name and the headline (first line) have to match in QOwnNotes. Do you want them to be different from each other?

**Comment by juranta** [comment]
Yes, I'd like them to be different. The file name (shown in left panel) is kind of a title of my notes and then the body of the note is something else. Sometimes it'll have a top header, sometimes it's just text, etc. I tried to import my notes to QOwnNotes and first the left panel showed the file names but they weren't searchable. When I tried to edit notes I noticed it tried to change the file names to match headers. It's perhaps a nice feature if all notes are like that.

**Comment by pbek** [comment]
Currently that is needed to be compatible with the Notes app of ownCloud and it's hardcoded on many places. I will keep your request as reference for the future, but close it for now. Maybe I can do something about it some day.

**Comment by pbek** [comment]
I will evaluate how much work it is to add a new, optional behavior.

**Comment by pbek** [comment]
In the just released version 16.05.5...

- you can new use a new setting in the general settings to **allow that the 
  note file name is different from the note headline**
    - this allows you to keep your old note files regardless of what the content
      of them is
    - the note name will also be found in the note search
    - you are able to **rename your note files** by double-clicking on the note 
      name in the note list
        - renaming will fail if there already exists a note with the same name
        - some characters are also not allowed in the file name, they will be 
          removed automatically
    - be aware of that ownCloud notes needs the note file name to be the same 
      as the note headline

Can you please test this new feature and report back?

**Comment by juranta** [comment]
Thanks! I'll test and report back next week when I get back to work.

**Comment by pbek** [comment]
Ok, thank you!

**Comment by juranta** [comment]
I've used this new option and it seems to work well with my existing database of notes. This mode may also work better for people who want to sync the notes with Dropbox and use other similar software on OS X, iOS, etc. For instance, iOS has lots of Dropbox syncing Markdown/plain text editors or note taking software. 

For my own use, I don't need header row when creating a new note either in this mode, but it's not a big thing. I can clear the header, or maybe learn to use it.

Thanks! I've tried some plain text notes programs like this for Windows, but they weren't good enough for my use - for instance, they often wouldn't work very well with external editors or sync, etc. I've used Sublime Text to edit my plain text notes on Windows. QOwnNotes is a very good Notational Velocity kind of an notes editor.
